pkg/cache/cluster.go

```diff
@@ -1007,6 +1007,106 @@ func (c *clusterCache) IterateHierarchy(key kube.ResourceKey, action func(resour
 	}
 }
 
+// IterateHierarchy iterates resource tree starting from the specified top level resources and executes callback for each resource in the tree
+func (c *clusterCache) IterateHierarchyV2(keys []kube.ResourceKey, action func(resource *Resource, namespaceResources map[kube.ResourceKey]*Resource) bool) {
```

Comment:
*sigh*, not sure what's up with this diff. Gonna try to fix the history real quick...